File: packages/firestore/ios/Plugin/Classes/Constraints/QueryLimitConstraint.swift
import Foundation
import FirebaseFirestore
import Capacitor

@objc public class QueryLimitConstraint: NSObject, QueryNonFilterConstraint {
    private var type: String
    private var limit: Int

    public init(_ queryConstraint: JSObject) {
        self.type = queryConstraint["type"] as! String
        self.limit = queryConstraint["limit"] as! Int
    }

    public func toQuery(query: Query) async throws -> Query {
        switch self.type {
        case "limit":
            return query.limit(to: self.limit)
        default:
            return query.limit(toLast: self.limit)
        }
    }
}

File: packages/firestore/android/src/main/java/app/capgo/capacitor/firebase/firestore/QueryLimitConstraint.java
package app.capgo.capacitor.firebase.firestore.classes.constraints;

import app.capgo.capacitor.firebase.firestore.interfaces.QueryNonFilterConstraint;
import com.getcapacitor.JSObject;
import com.google.firebase.firestore.Query;
import org.json.JSONException;

public class QueryLimitConstraint implements QueryNonFilterConstraint {

    private String type;
    private int limit;

    public QueryLimitConstraint(JSObject queryConstraint) throws JSONException {
        this.type = queryConstraint.getString("type");
        this.limit = queryConstraint.getInt("limit");
    }

    public int getLimit() {
        return limit;
    }

    public Query toQuery(Query query, com.google.firebase.firestore.FirebaseFirestore firestoreInstance) {
        if (type.equals("limit")) {
            return query.limit(limit);
        } else {
            return query.limitToLast(limit);
        }
    }
}
